src/PlatformEngines.jl

```diff
@@ -55,7 +55,9 @@ function get_server_dir(
         return
     end
     isempty(Base.DEPOT_PATH) && return
-    joinpath(depots1(), "servers", String(m.captures[1]))
+    invalid_filename_chars = [':', '/', '<', '>', '"', '/', '\\', '|', '?', '*']
+    dir = replace(String(only(m)), (c => '_' for c in invalid_filename_chars)...);
```

Comment:
This doesn't work on 1.6 (just saw the backport-1.6 label).

```suggestion
    dir = join(replace(c -> c in invalid_filename_chars ? '_' : c, collect(String(only(m)))))
```
should work, but is also pretty ugly.